.github/workflows/workflow.yml

```diff
@@ -0,0 +1,34 @@
+name: Run unit tests
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [assigned, opened, reopened, ready_for_review, synchronize]
+  workflow_dispatch:
+    workflows: ["Run tests"]
+    branches: [main]
```

Comment:
You need to specify the correct default branch for this, but IMO this shouldn't be enforced. If you enforce workflow_dispatch for a specific branch, you can't retrigger it on other branches.

Reply:
But the other branches will run with `pull_request`.

Reply:
I mean, I guess I could generalize this for situations where the CI gets stuck. That way there is no need to force push only to re-run the CI.

Reply:
> I mean, I guess I could generalize this for situations where the CI gets stuck. That way there is no need to force push only to re-run the CI.

Yes, that's it